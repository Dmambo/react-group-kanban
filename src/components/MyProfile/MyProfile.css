.myprofile {
  display: flex;

  /* justify-content: space-evenly; */
  align-items: center;
  margin: 10px 50px;
  gap: 60px;
}

.profile_item {
  display: flex;
  flex-direction: column;
  list-style: none;
  border: 1px solid #0000009a;
  margin: 0;
  padding-inline-start: 0;
  border-radius: 5px;
}

.profile_title {
  font-size: 2.2rem;
}

.profile_list {
  width: 600px;
  padding: 20px;
  border: 1px solid #0000009a;
  display: flex;
  align-items: center;
  margin: 0;
}
