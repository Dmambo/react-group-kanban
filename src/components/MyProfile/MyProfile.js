import React from 'react';
import { useSelector } from 'react-redux';
import './MyProfile.css';

const MyProfile = () => {
  const missions = useSelector((state) => state.missionsReducer.missions);
  const rockets = useSelector((state) => state.rockets);
  const reservedMissions = Array.isArray(missions)
    ? missions.filter((mission) => mission.reserved)
    : [];
  const reservedRockets = Array.isArray(rockets)
    ? rockets.filter((rocket) => rocket.reserved)
    : [];
  return (
    <div className="myprofile">
      <div className="myprofile__missions">
        <h2 className="profile_title">My Missions</h2>
        <ul className="profile_item">
          {reservedMissions.map((mission) => (
            <li key={mission.mission_id} className="profile_list">
              {mission.mission_name}
              {' '}
            </li>
          ))}
        </ul>
      </div>
      <div className="myprofile__rockets">
        <h2 className="profile_title">My Rockets</h2>
        <ul className="profile_item">
          {reservedRockets.map((rocket) => (
            <li key={rocket.id} className="profile_list">
              {rocket.name}
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};
export default MyProfile;
