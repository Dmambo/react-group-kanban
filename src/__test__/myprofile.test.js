import React from 'react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import renderer from 'react-test-renderer';
import MyProfile from '../components/MyProfile/MyProfile';

describe('Test MyProfile Component', () => {
  it('should render MyProfile component', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MyProfile />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
