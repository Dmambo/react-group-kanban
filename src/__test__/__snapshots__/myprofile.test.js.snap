// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MyProfile Component should render MyProfile component 1`] = `
<div
  className="myprofile"
>
  <div
    className="myprofile__missions"
  >
    <h2
      className="profile_title"
    >
      My Missions
    </h2>
    <ul
      className="profile_item"
    />
  </div>
  <div
    className="myprofile__rockets"
  >
    <h2
      className="profile_title"
    >
      My Rockets
    </h2>
    <ul
      className="profile_item"
    />
  </div>
</div>
`;
